tests/integration/scripts/test_create_dataset.py

```diff
@@ -0,0 +1,49 @@
+import os
+import tempfile
+from pathlib import Path
+
+import pytest
+
+from maze_transformer.training.mazedataset import MazeDataset
+from scripts.create_dataset import create_dataset
+
+
+@pytest.fixture
+def temp_dir() -> Path:
+    data_dir = tempfile.TemporaryDirectory()
+    yield Path(data_dir.name)
+    data_dir.cleanup()
+
+
+def test_create_expected_files_and_directories(temp_dir):
+    n_mazes = 5
+    grid_n = 3
+    name = "test"
+
+    create_dataset(path_base=str(temp_dir), n_mazes=n_mazes, grid_n=grid_n, name=name)
+    dataset_directory_name = f"g{grid_n}-n{n_mazes}-{name}"
+    files = MazeDataset.DISK_SAVE_FILES
+    file_names = {
+        value for attr, value in vars(files).items() if not attr.startswith("__")
+    }
```

Comment:
super unimportant, but I actually think it would be nicer to inline the variable

Reply:
It's the autoformatter 😂